File: GuideObjC/GuideObjC/Views/ProgressCell/ProgressCell.m
//
//  ProgressCell.m
//  GuideObjC
//
//  Created by LuzanovRoman on 21.12.16.
//  Copyright © 2016 eigengraph. All rights reserved.
//

#import "ProgressCell.h"

@interface ProgressCell ()
@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *activityIndicatorView;
@end

@implementation ProgressCell

- (void)setIndicatorIsHidden:(BOOL)indicatorIsHidden {
    _indicatorIsHidden = indicatorIsHidden;
    _activityIndicatorView.hidden = indicatorIsHidden;
}

- (void)setSelected:(BOOL)selected {
    [super setSelected:selected];
    [_activityIndicatorView startAnimating];
}
@end

File: GuideSwift/GuideSwift/Views/ProgressCell/ProgressCell.swift
//
//  ProgressCell.swift
//  GuideSwift
//
//  Created by LuzanovRoman on 21.12.16.
//  Copyright © 2016 eigengraph. All rights reserved.
//

import UIKit

class ProgressCell: UITableViewCell {
    @IBOutlet weak var activityIndicatorView: UIActivityIndicatorView!

    var indicatorIsHidden: Bool = true {
        didSet {
            activityIndicatorView.isHidden = indicatorIsHidden
        }
    }

    override public var isSelected: Bool {
        get {
            return super.isSelected
        }
        set {
            super.isSelected = newValue
            activityIndicatorView.startAnimating()
        }
    }
}
